wgkex/worker/netlink.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+import re
+import hashlib
+
+from salt.utils.network import mac2eui64
+from textwrap import wrap
+from typing import Dict, List
+from pyroute2 import WireGuard, IPRoute
+from pyroute2.netlink.rtnl import ndmsg
+from typing import Dict, List
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+
+
+@dataclass
+class WireGuardClient:
+    """WireGuardClient describes complete configuration for a specific WireGuard client
+
+    Attributes:
+        public_key: WireGuard Public key
+        domain: Domain Name of the WireGuard peer
+        lladdr: IPv6 lladdr of the WireGuard peer
+        wg_interface: Name of the WireGuard interface this peer will use
+        vx_interface: Name of the VXLAN interface we set a route for the lladdr to
+        remove: Are we removing this peer or not?
+    """
+
+    public_key: str
+    domain: str
+    lladdr: bytes
+    wg_interface: str
+    vx_interface: str
+    remove: bool
+
+
+# we receive stuff from wgkex-broker
+def generate_lladdr(public_key: str) -> str:
+    m = hashlib.md5()
+
+    m.update(public_key.encode("ascii") + b"\n")
+    hashed_key = m.hexdigest()
+    hash_as_list = wrap(hashed_key, 2)
+    temp_mac = ":".join(["02"] + hash_as_list[:5])
+
+    lladdr = re.sub("\/\d+$", "/128", mac2eui64(mac=temp_mac, prefix="fe80::/10"))
+    return lladdr
+
+
+def generate_interface_names(peer: WireGuardClient) -> WireGuardClient:
+    peer.wg_interface = "wg-" + peer.domain
+    peer.vx_interface = "vx-" + peer.domain
+
+    return peer
+
+
+def cleanup_wireguard_clients(domain: str) -> bool:
+    stale_clients = find_stale_wireguard_clients("wg-" + domain)
+    result = []
+    for stale_client in stale_clients:
+        stale_wireguard_client = WireGuardClient(
+            public_key=stale_client,
+            lladdr=generate_lladdr(stale_client),
+            domain=domain,
+            wg_interface="",
+            vx_interface="",
+            remove=True,
+        )
+        stale_wireguard_client = generate_interface_names(stale_wireguard_client)
+        result = link_handler(stale_wireguard_client)
+    return result
+
+
+# pyroute2 stuff
+def link_handler(client: WireGuardClient) -> Dict:
+    results = {}
+
+    results.update({"Wireguard": wireguard_handler(client)})
+    try:
+        results.update({"Route": route_handler(client)})
+    except Exception as e:
+        results.update({"Route": e})
+    results.update({"Bridge FDB": bridge_fdb_handler(client)})
+
+    return results
+
+
+def bridge_fdb_handler(client: WireGuardClient) -> Dict:
+    ip = IPRoute()
+
+    action = "append"
+    if client.remove:
+        action = "del"
+
+    result = ip.fdb(
+        action,
+        ifindex=ip.link_lookup(ifname=client.vx_interface)[0],
+        lladdr="00:00:00:00:00:00",
+        dst=re.sub("\/\d+$", "", client.lladdr),
+    )
+
+    ip.close()
+    return(result)
```

Comment:
```suggestion
    return result
```
remove redundant parenthesis